encoder/vaapiencoder_h264.cpp

```diff
@@ -1516,21 +1627,60 @@ bool VaapiEncoderH264::fill(VAEncPictureParameterBufferH264* picParam, const Pic
     picParam->pic_fields.bits.transform_8x8_mode_flag = m_videoParamAVC.enableDct8x8;
     picParam->pic_fields.bits.deblocking_filter_control_present_flag = true;
 
+    picParam->pic_fields.bits.pic_scaling_matrix_present_flag =
+        m_videoParamAVC.qMatrix.picScalingMatrixPresentFlag;
+
     return TRUE;
 }
 
+bool VaapiEncoderH264::fill(VAIQMatrixBufferH264* buffer) const
+{
+    uint8_t *dest = (uint8_t *)buffer;
+    const VideoParamsQM *qMatrix = &m_videoParamAVC.qMatrix;
+    const uint8_t *list_present_flag;
+    int i;
+
+    if (qMatrix->picScalingMatrixPresentFlag) {
+        list_present_flag = qMatrix->picScalingListPresentFlag;
+    } else if (qMatrix->seqScalingMatrixPresentFlag) {
+        list_present_flag = qMatrix->seqScalingListPresentFlag;
+    } else {
+        return true;
```

Comment:
This will not happen because I have checked this before calling fill():
    if (!qMatrix->picScalingMatrixPresent &&
        !qMatrix->seqScalingMatrixPresent ) {
        return true;
    }